agent/provider/tests/keystore_cli.rs

```diff
@@ -145,117 +145,189 @@ fn test_keystore_list_cmd_creates_cert_dir_in_dir_set_by_arg() {
     vec![("fe4f04e2", "none"), ("55e451bd", "outbound-manifest"), ("20e9eb45", "outbound-manifest")];
     "Add multiple certificates and check permissions"
 )]
+#[test_case(
+    vec!["root-certificate.signed.json", "partner-certificate.signed.json"],
+    vec![],
+    false,
+    vec![("80c84b27", "all"), ("cb16a2ed", "{\"outbound\":\"unrestricted\"}")];
+    "Add multiple Golem certificates and check permissions"
+)]
+#[test_case(
+    vec!["foo_req.cert.pem", "partner-certificate.signed.json"],
+    vec![],
+    false,
+    vec![("25b9430c", "none"), ("cb16a2ed", "{\"outbound\":\"unrestricted\"}")];
+    "Add Golem and X509 certificate and then check permissions"
+)]
 #[serial]
 fn test_keystore_add_certificate_permissions(
     certificates: Vec<&str>,
     permissions: Vec<&str>,
     whole_chain: bool,
     expected: Vec<(&str, &str)>,
 ) {
+    let result = add_and_list(certificates, permissions, whole_chain);
+    for (cert_id, perm) in expected {
+        assert_eq!(read_permissions(&result, dbg!(cert_id)), perm);
+    }
+}
+
+#[serial]
+#[test]
+fn test_keystore_add_certificate_second_time() {
     let (resource_cert_dir, cert_dir) = CERT_TEST_RESOURCES.init_cert_dirs();
 
-    let mut command = Command::cargo_bin("ya-provider").unwrap();
-    command
-        .args(["--cert-dir", cert_dir.as_path().to_str().unwrap()])
-        .arg("keystore")
-        .arg("add");
+    add(
+        vec!["foo_ca-chain.cert.pem"],
+        vec!["all"],
+        true,
+        &resource_cert_dir,
+        &cert_dir,
+    );
 
-    if !certificates.is_empty() {
-        for certificate in certificates {
-            command.arg(resource_cert_dir.join(certificate));
-        }
-    }
+    // This call doesn't specify any permissions, so these should be removed.
+    add(
+        vec!["foo_ca-chain.cert.pem"],
+        vec![],
+        true,
+        &resource_cert_dir,
+        &cert_dir,
+    );
 
-    if !permissions.is_empty() {
-        command.arg("--permissions");
-        permissions.iter().for_each(|permission| {
-            command.arg(permission);
-        })
-    }
+    let result = list_certificates_command(&cert_dir).unwrap();
 
-    if whole_chain {
-        command.arg("--whole-chain");
-    }
+    assert_eq!(read_permissions(&result, "55e451bd"), "none");
+    assert_eq!(read_permissions(&result, "fe4f04e2"), "none");
+}
 
-    command.arg("--json").assert().success();
+#[serial]
+#[test]
+fn test_keystore_remove_certificate_check_permissions() {
+    let (resource_cert_dir, cert_dir) = CERT_TEST_RESOURCES.init_cert_dirs();
+
+    add(
+        vec!["foo_ca-chain.cert.pem", "foo_req.cert.pem"],
+        vec!["outbound-manifest"],
+        true,
+        &resource_cert_dir,
+        &cert_dir,
+    );
+
+    // This call doesn't specify any permissions, so the will be removed.
```

Comment:
This sentence needs to be redacted somehow

Reply:
Updated.